domain/machine/modelmigration/export.go

```diff
@@ -22,34 +24,97 @@ func RegisterExport(coordinator Coordinator, logger logger.Logger) {
 	})
 }
 
-// ExportService provides a subset of the machine domain
-// service methods needed for machine export.
+// ExportService defines the machine service used to export machines to
+// another controller model to this controller.
 type ExportService interface {
+	// AllMachineNames returns the names of all machines in the model.
 	AllMachineNames(ctx context.Context) ([]coremachine.Name, error)
+	// InstanceID returns the cloud specific instance id for this machine.
+	// If the machine is not provisioned, it returns a NotProvisionedError.
+	InstanceID(ctx context.Context, machineUUID string) (string, error)
```

Comment:
We already pass UUIDs as strings on all (most?) of the domains. I personally prefer this approach instead of having a (probably just alias) type on core and a new dependency.

Reply:
Fair enough, I don't see indisputable gain to have UUID type, at least in the short term, but I can see how painful it would already be to align.